src/telemetry_poller_sup.erl

```diff
@@ -0,0 +1,24 @@
+%% @private
+-module(telemetry_poller_sup).
+
+-behaviour(supervisor).
+
+-export([start_link/1]).
+-export([init/1]).
+
+-define(SERVER, ?MODULE).
+
+start_link(Opts) ->
+    supervisor:start_link({local, ?SERVER}, ?MODULE, Opts).
+
+init(Opts) ->
+    SupFlags = #{strategy => one_for_one,
+                 intensity => 1,
+                 period => 5},
+    Poller = #{id => telemetry_poller,
```

Comment:
You could call `telemetry_poller:child_spec(Arg)`. This would be a good way to test that our `child_spec/1` function is correct.

Reply:
Great suggestion!